tests/unit/core/exporter/exportable_code/demo/demo_package/executors/test_asynchronous.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Test of AsyncExecutor in demo_package."""
+
+from unittest.mock import MagicMock
+
+import pytest
+from otx.core.exporter.exportable_code.demo.demo_package.executors import asynchronous as target_file
+from otx.core.exporter.exportable_code.demo.demo_package.executors.asynchronous import AsyncExecutor
+
+
+class MockAsyncPipeline:
+    def __init__(self, *args, **kwargs):
+        self.arr = []
+        self.idx = 0
+
+    def get_result(self, *args, **kwawrgs):
+        if self.idx >= len(self.arr):
+            return None
+        ret = self.arr[self.idx]
+        self.idx += 1
+        return ret
+
+    def submit_data(self, frame, *args, **kwawrgs):
+        self.arr.append(frame)
+
+    def await_all(self):
+        pass
+
+
+class TestAsyncExecutor:
+    @pytest.fixture(autouse=True)
+    def setup(self, mocker):
+        mocker.patch.object(target_file, "AsyncPipeline", side_effect=MockAsyncPipeline)
+
+    @pytest.fixture()
+    def mock_model(self):
+        return MagicMock()
+
+    @pytest.fixture()
+    def mock_visualizer(self):
+        visualizer = MagicMock()
+        visualizer.is_quit.return_value = False
+        visualizer.draw.side_effect = lambda x, _: x
+        return visualizer
+
+    def test_init(self, mock_model, mock_visualizer):
+        AsyncExecutor(mock_model, mock_visualizer)
+
+    @pytest.fixture()
+    def mock_streamer(self, mocker):
+        return mocker.patch.object(target_file, "get_streamer", return_value=range(1, 4))
+
+    @pytest.fixture()
+    def mock_dump_frames(self, mocker):
+        return mocker.patch.object(target_file, "dump_frames")
+
+    def test_run(self, mocker, mock_model, mock_visualizer, mock_streamer, mock_dump_frames):
+        mock_render_result = mocker.patch.object(AsyncExecutor, "render_result", side_effect=lambda x: x)
+        executor = AsyncExecutor(mock_model, mock_visualizer)
+        executor.run(MagicMock())
+
+        mock_render_result.assert_called()
+        for i in range(1, 4):
+            assert mock_render_result.call_args_list[i - 1].args == (i,)
+        mock_visualizer.show.assert_called()
+        for i in range(1, 4):
+            assert mock_visualizer.show.call_args_list[i - 1].args == (i,)
+        mock_dump_frames.assert_called()
```

Comment:
You are testing `AsyncExecutor` and it consists of three components: 1) `ModelWrapper`, 2) `BaseVisualizer`, and 3) `AsyncPipeline`. In this case, we basically mock those three components to test the target object. But, you mocked two object, but rather than mocking `AsyncPipeline`, trying to mock the member function of the target test object. It is not common practice. The common practice will be
```suggestion
    def test_run(self, mocker, mock_model, mock_visualizer, mock_streamer, mock_dump_frames):
        mock_async_pipeline = mocker.patch("otx.core.exporter.exportable_code.demo.demo_package.executors.asynchronous.AsyncPipeline").return_value
        executor = AsyncExecutor(mock_model, mock_visualizer)
        executor.run(MagicMock())

        ...
```

Reply:
I also mock `AsyncPipeline` and `render_result` as you said. I think purpose of unit test is to check target function works as expected. So, checking how `render_result` works is out of scope of those test function. I also implement unit test for `render_result` which can cover checking `render_result`, I think.

Reply:
I am still feeling the idea of ​​mocking an object's "public" function in order to test itself very strange. It deviates from the standards of software development. Thus, it makes other developers hard to understand it.

We are not just copy pasting something from OTX 1.x to OTX 2. We are doing this to achieve real improvement.

This time I'll approved this PR, However, please think about it next time.